api/models/Post.js

```diff
@@ -362,6 +362,26 @@ module.exports = bookshelf.Model.extend(Object.assign({
     PUBLIC_READABLE: 1
   },
 
+ projectsForContext: async function (context, contextId) {
+    let postIds
+    
+    switch (context) {
+      default: // For now, the only context is member profiles. Add to this once we show projects on group landing pages.
+        postIds = await this.query(q => {
+          q.select('posts.*')
+          q.from('group_memberships')
+          q.leftJoin('groups', 'group_memberships.group_id', 'groups.id')
+          q.leftJoin('posts', 'groups.group_data_id', 'posts.id')
+          q.whereNotNull('group_memberships.project_role_id')
+          q.andWhere('group_memberships.user_id', contextId)
+          q.andWhere('group_memberships.active', true)
+        }).query()
+          .then(rows => rows.map(r => r.id))
```

Comment:
why are you doing multiple queries, first getting ids then loading those ids again?

Reply:
Had same question on this one.

Reply:
I didn't realize you could do `this.query(.....).fetchAll()` - will update